plugins/net_plugin/net_plugin.cpp

```diff
@@ -1511,6 +1552,39 @@ namespace eosio {
       sync_wait();
    }
 
+   //-----------------------------------------------------------
+   void block_status_monitor::accepted() {
+      if( window_start_ != fc::time_point() ) {
+         window_start_ = fc::time_point();
+         events_ = 0;
+      }
+   }
+
+   void block_status_monitor::rejected() {
+      const auto now = fc::time_point::now();
+      // if starting a new window
+      if( window_start_ == fc::time_point() ) {
+         window_start_ = now;
+         return;
```

Comment:
Seems to me if you are starting a new window you should increment events_.
I think it would be clearer to use `if( events_ == 0 ) {` here.

Reply:
Changed logic to make it clearer